appinventor/blocklyeditor/src/workspace_svg.js

```diff
@@ -374,6 +374,38 @@ Blockly.WorkspaceSvg.prototype.getProcedureDatabase = function() {
   return this.procedureDb_;
 };
 
+//noinspection JSUnusedGlobalSymbols Called from BlocklyPanel.java
+/**
+ * Adds a screen name to the list tracked by the workspace.
+ * @param {string} name The name of the new screen.
+ */
+Blockly.WorkspaceSvg.prototype.addScreen = function(name) {
+  if (this.screenList_.indexOf(name) == -1) {
+    this.screenList_.push(name);
+    this.typeBlock_.needsReload.screens = true;
+  }
+};
+
+//noinspection JSUnusedGlobalSymbols Called from BlocklyPanel.java
+/**
+ * Removes a screen name from the list tracked by the workspace.
+ * @param {string} name The name of the screen to remove.
+ */
+Blockly.WorkspaceSvg.prototype.removeScreen = function(name) {
+  if (this.screenList_.indexOf(name) != -1) {
```

Comment:
One minor optimization would be to store the value of the indexOf call and then reuse it within the if statement.